### ISSUE TITLE
[chip dv] Do not use cfg.clk_rst_vif.wait_clks() to wait for clock cycles

### ISSUE THREAD

**Issue by sriyerg** [lowRISC/opentitan#14918]
We need to replace the `cfg.clk_rst_if.wait_clks()` invocations in our test sequences with the appropriate clk / rst monitor in `chip_if ` for example, `cfg.chip_vif.sys_clk_rst_if.wait_clks()`.

The reason for this is explained in this commit: https://github.com/lowRISC/opentitan/pull/14917/commits/bb2bb8abf3cefcb5bb0b25680b6c74051809e5cb

Previously, the `wait_clks` on this would wait on the external clock source, which was unsed in most tests. As of the commit above, the clock port of this interface is unconnected. The clock remains active to prevent breakages.

**Comment by sriyerg** [comment]
@weicaiyang, FYI.

**Comment by weicaiyang** [comment]
SGTM

**Comment by tjaychen** [comment]
this is probably something that can't be deferred to M3 right? It sounds like if we don't fix this we might see random errors

**Comment by sriyerg** [comment]
AFAICT, there have not been any breakages related to this yet. Maybe we can defer to M3 if tests are largely passing. Its just that the DV sequences are waiting on clock events using a clock that has nothing to do with the chip.

**Comment by tjaychen** [comment]
that's kind of surprising there are no failures... alright, maybe let's defer like you said and see if anything catches fire hehe.

**Comment by weicaiyang** [comment]
most of the wait_clk just wants to add some delays. How big the delay is may not really matter.
For things we really care, we should make it wait for an event.

**Comment by tjaychen** [comment]
wait but i thought the issue was that the "clk" we're sitting on may not always be there, so if we wait on that clock, things might just stall?

**Comment by weicaiyang** [comment]
> wait but i thought the issue was that the "clk" we're sitting on may not always be there, so if we wait on that clock, things might just stall?

`cfg.clk_rst_if` is a TB clock, which is always on but not used by DUT. I think Sri's point is to replace it with an actually design clock like sys_clk.

**Comment by tjaychen** [comment]
ah sounds good, i think i misunderstood it to mean the clock was not always on.
